src/main/java/com/paskar/email/application/model/Role.java

```diff
@@ -0,0 +1,25 @@
+package com.paskar.email.application.model;
+
+import lombok.Getter;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public enum Role {
+    USER(Set.of(Permission.WRITE)),
+    ADMIN(Set.of(Permission.READ_DELETE, Permission.WRITE));
+
+    Role(Set<Permission> permissions) {
+        this.permissions = permissions;
+    }
+
+    @Getter
+    private final Set<Permission> permissions;
+
+    public Set<SimpleGrantedAuthority> getAuthorities(){
+        return getPermissions().stream()
+                .map(permission -> new SimpleGrantedAuthority(permission.getPermission()))
+                .collect(Collectors.toSet());
```

Comment:
java.util.stream.Collectors#toUnmodifiableSet

I do not think that you need possibility to add something new to Set.